source/adam_evaluate.cpp

```diff
@@ -92,11 +90,14 @@ namespace adobe {
 adam_callback_suite_t bind_to_sheet(sheet_t& sheet) {
     adam_callback_suite_t suite;
 
-    suite.add_cell_proc_m = boost::bind(&add_cell, boost::ref(sheet), _1, _2, _3, _4);
-    suite.add_relation_proc_m = boost::bind(&add_relation, boost::ref(sheet), _1, _2, _3, _4);
-    suite.add_interface_proc_m =
-        boost::bind(&adobe::sheet_t::add_interface, boost::ref(sheet), _1, _2, _3, _4, _5, _6);
-
+    suite.add_cell_proc_m = std::bind(&add_cell, std::ref(sheet), _1, _2, _3, _4);
+    suite.add_relation_proc_m = std::bind(&add_relation, std::ref(sheet), _1, _2, _3, _4);
+    suite.add_interface_proc_m = [&sheet](name_t name, bool linked, const line_position_t& position1,
+                                         const array_t& initializer, const line_position_t& position2,
+                                         const array_t& expression, const std::string& brief,
+                                         const std::string& detailed) -> void {
+        sheet.add_interface(name, linked, position1, initializer, position2, expression);
```

Comment:
There is a mismatch between signatures of `add_interface_proc_m` and `sheet_t::add_interface`. While former has 8 args, later has only 6.
```
    using add_interface_proc_t = std::function<void(
        name_t cell_name, bool linked, const line_position_t& position1, const array_t& initializer,
        const line_position_t& position2, const array_t& expression, const std::string& brief,
        const std::string& detailed)>;
```
```
void sheet_t::add_interface(name_t name, bool linked, const line_position_t& position1,
                            const array_t& initializer, const line_position_t& position2,
                            const array_t& expression)
```

Reply:
I believe boost bind just drops the extra arguments. If`std::bind` behaves differently, I'm fine with the lambda here (the two missing arguments are for comments from the parser but the sheet doesn't do anything with them.